networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -123,24 +123,15 @@ def get_vrf_context(self):
         else:
             return None
 
-    @property
-    def get_vrf_details(self):
-        if self.apic_driver:
-            return self._get_vrf_details
-        else:
-            return None
-
     def _get_vrf_context_gbp(self, context, router_id, port_db):
-        l2p = self.apic_driver._network_id_to_l2p(
-            context, port_db['network_id'])
-        l3p = self.apic_driver.gbp_plugin.get_l3_policy(
-            context, l2p['l3_policy_id'])
-        return {'vrf_id': l3p['id']}
-
-    def _get_vrf_context_neutron(self, context, router_id, port_db):
-        return {'router_id': router_id}
-
-    def _get_vrf_details_gbp(self, context, **kwargs):
+        router = self.l3_plugin.get_router(context, router_id)
+        fixed_ips = router[l3.EXTERNAL_GW_INFO]['external_fixed_ips']
+        # TODO(tbachman): Do we need to handle multiple fixed IPs?
+        # If so, we should match up the subnet_id name with the APIC
+        # owned prefix, as there should only be one of them
+        es_list = self.apic_driver.gbp_plugin.get_external_segments(
+            context.elevated(), {'subnet_id': [fixed_ips[0]['subnet_id']]})
+        kwargs = {'vrf_id': es_list[0]['l3_policies'][0]}
```

Comment:
its possible a ES is associated with multiple l3Ps so this won't work I think...